vertx-sql-client/src/test/java/io/vertx/sqlclient/benchmarks/AccumulatorReadBenchmark.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2011-2022 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.sqlclient.benchmarks;
+
+import io.vertx.sqlclient.impl.accumulator.Accumulator;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.infra.Blackhole;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import static io.vertx.sqlclient.benchmarks.Utils.generateStrings;
+
+@Threads(1)
+@State(Scope.Thread)
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@Warmup(iterations = 20, time = 1, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 10, time = 2, timeUnit = TimeUnit.SECONDS)
+@Fork(value = 1, jvmArgs = {"-Xms8g", "-Xmx8g", "-Xmn7g"})
+public class AccumulatorReadBenchmark {
+
+  @Param({"ARRAY_LIST", "CHUNKED_FIXED_SIZE", "CHUNKED_GROWING_SIZE"})
+  public AccumulatorType accumulatorType;
+
+  @Param({"5", "20", "65", "605", "1820", "5465", "16400"})
+  int size;
+
+  @Param({"false", "true"})
+  boolean shuffle;
+
+  @Param({"false", "true"})
+  boolean gc;
+
+  String[] arr;
+  Accumulator<String> accumulator;
+
+  @Setup
+  public void setup() throws IOException, InterruptedException {
+    arr = generateStrings(size, shuffle, gc);
+    accumulator = accumulatorType.newInstance();
+    for (String s : arr) {
+      accumulator.accept(s);
+    }
+  }
+
+  @Benchmark
+  public void iterate(Blackhole bh) {
+    for (String s : accumulator) {
+      bh.consume(s.hashCode());
```

Comment:
you can just consume `s`?

Reply:
or you can do something better here: `BlackHole::consume` preven some JIT loop transformations that are key to compare between versions:

```java
  @Benchmark
  public int iterate() {
    int dummy = 0;
    for (String s : accumulator) {
      dummy += s.length();
    }
    return dummy;
}      
```
This will make the loop as fast as it can as in the real world and will prevent it to be dead-code-eliminated

I suggest to re-run the bench after changing this